pulser-core/pulser/sequence/sequence.py

```diff
@@ -376,74 +377,130 @@ def switch_device(
+                        phase_jump_time_check = (
+                            od_ch_obj.phase_jump_time
+                            == nd_ch_obj.phase_jump_time
+                        )
+                        clock_period_check = (
+                            nd_ch_obj.clock_period % od_ch_obj.clock_period
+                            == 0
+                        )
+
+                        if not phase_jump_time_oracle:
+                            if phase_jump_time_check and clock_period_check:
+                                channel_matching[channel_name] = nd_ch_id
+                                matched = True
+                                break
+                            else:
+                                matched = False
                                 strict_error_message = (
-                                    "No phase_jump_time & clock_period match."
+                                    "No channel with phase_jump_time"
+                                    + " & clock_period match."
                                 )
+                        else:
+                            if clock_period_check:
+                                channel_matching[channel_name] = nd_ch_id
+                                matched = True
+                                alert1 = phase_jump_time_check is False
+                                break
                             else:
-                                strict_error_message = None
-
-                        if strict_error_message is None:
-                            sw_channel_args = list(call.args)
-                            # Switch the old id with the correct id
-                            sw_channel_args[1] = nd_ch_id
-                            getattr(new_seq, "declare_channel")(
-                                *sw_channel_args, **call.kwargs
-                            )
-                            called = True
-                            break
-                    elif basis_match and not addressing_match:
-                        ch_type_er_mess = "No addressing match."
-                    elif not basis_match and addressing_match:
-                        ch_type_er_mess = "No basis match."
+                                matched = False
+                                strict_error_message = (
+                                    "No channel with phase_jump_time"
+                                    + " & clock_period match."
+                                )
                     else:
-                        ch_type_er_mess = "No basis and addressing match."
+                        channel_matching[channel_name] = nd_ch_id
+                        matched = True
+                        break
+                else:
+                    matched = False
+                    ch_type_er_mess = (
+                        "No channel with both basis and addressing match."
+                    )
```

Comment:
This block also seems like it can be simplified. There is some unnecessary code replication. For example, there is no need to check `phase_jump_time_oracle` if `phase_jump_time_check is True`

Reply:
The oracle is there to identify whether it is relevant to take the phase_jump_time value. If it is false we don't care of phase_jump_time value, either False or True.